src/core/thread/mle.cpp

```diff
@@ -1971,6 +1971,11 @@ otError Mle::SendDataRequest(const Ip6::Address &aDestination,
         message->Free();
     }
 
+    if (!IsRxOnWhenIdle())
+    {
+        mChildUpdateRequestTimer.Start(kMaxChildUpdateResponseTimeout);
```

Comment:
Do we need to stop the timer from `HandleDataResponse()`?
(I think we should stop it in case of successful processing of the received response and only if `!IsRxOnWhenIdle()`).

Reply:
Yes, we should.  Added a call to `mChildUpdateRequestTimer.Stop()` in `Mle::HandleLeaderData()`.